Change the HTTP Edge "encoder" to a "decoder".

mreid-moz opened the pull request:
This makes far more sense logically, but it's a bit less convenient
since you have to manually copy the incoming message in the decoder.

This enables outputs to use message matchers to properly route data.

mreid-moz committed:
Change the HTTP Edge "encoder" to a "decoder".

This makes far more sense logically, but it's a bit less convenient
since you have to manually copy the incoming message in the decoder.

This enables outputs to use message matchers to properly route data.

trink commented:
Do you want the PR that would avoid the manual copy?

mreid-moz commented:
Nah, let's get this in first, and clean it up when the improved message API PR lands.

trink commented on heka/sandbox/decoders/http_edge_decoder.lua:
```diff
@@ -62,20 +76,43 @@ local msg = {
+
     -- Reset Fields, since different namespaces may use different fields.
-    msg.Fields = {}
+    main_msg.Fields = {}
```
Why not just modify the landfill message here?

mreid-moz replied:
It seemed clearer to keep the messages separate.

trink replied:
Then I would at least move the data from the landfill message to the new message instead of hitting the CGO interface again.

mreid-moz committed:
Get field values from landfill_msg.

Get values from the local lua table rather than calling
"read_message(...)" again.
